src/pivot-table/components/cells/ColumnAdjuster.tsx

```diff
@@ -14,47 +13,41 @@ interface AdjusterProps {
   dataModel: DataModel | undefined;
   isLastColumn: boolean;
 }
-
 /**
  * Component that is placed on top of column border, to resize the columns.
  * When you start dragging, mouse move and mouse up listeners are added.
  * While dragging this components follows the pointer, and on mouse up all column widths are updated.
  */
 const ColumnAdjuster = ({ cellInfo, columnWidth, dataModel, isLastColumn }: AdjusterProps) => {
-  const [internalWidth, setInternalWidth] = useState(columnWidth);
-  const tempWidth = useRef({ initWidth: 0, columnWidth: 0, initX: 0 });
+  const [, setInternalWidth] = useState(columnWidth);
   const positionAdjustment = isLastColumn ? CELL_PADDING : CELL_PADDING + GRID_BORDER;
 
-  useOnPropsChange(() => {
-    setInternalWidth(columnWidth);
-  }, [columnWidth]);
+  const tempWidth = useMemo(() => ({ initWidth: 0, columnWidth, initX: 0 }), [columnWidth]);
 
   const mouseMoveHandler = (evt: MouseEvent) => {
-    const deltaWidth = evt.clientX - tempWidth.current.initX;
-    const adjustedWidth = Math.max(tempWidth.current.initWidth + deltaWidth, ColumnWidthValues.PixelsMin);
+    const deltaWidth = evt.clientX - tempWidth.initX;
+    const adjustedWidth = Math.max(tempWidth.initWidth + deltaWidth, ColumnWidthValues.PixelsMin);
     setInternalWidth(adjustedWidth);
-    tempWidth.current.columnWidth = adjustedWidth;
+    tempWidth.columnWidth = adjustedWidth;
   };
 
   const mouseUpHandler = (evt: MouseEvent) => {
     preventDefaultBehavior(evt);
     document.removeEventListener("mousemove", mouseMoveHandler);
     document.removeEventListener("mouseup", mouseUpHandler);
 
-    if (tempWidth.current.columnWidth !== tempWidth.current.initWidth) {
-      const newWidthData = { type: ColumnWidthType.Pixels, pixels: tempWidth.current.columnWidth };
+    if (tempWidth.columnWidth !== tempWidth.initWidth) {
+      const newWidthData = { type: ColumnWidthType.Pixels, pixels: tempWidth.columnWidth };
       dataModel?.applyColumnWidth(newWidthData, cellInfo);
     }
   };
 
   const mouseDownHandler = (evt: React.MouseEvent) => {
     evt.stopPropagation();
 
-    tempWidth.current = {
-      initX: evt.clientX,
-      initWidth: columnWidth,
-      columnWidth,
-    };
+    tempWidth.initX = evt.clientX;
+    tempWidth.initWidth = columnWidth;
```

Comment:
I don't think you need to assing `initWidth` and `columnWidth` if you set the values via useMemo instead:
`const tempWidth = useMemo(() => ({ initWidth: columnWidth, columnWidth, initX: 0 }), [columnWidth]);`

Reply:
we cant create  hook within the nested function so we cant do it

Reply:
See my other comment about this